galley/pkg/testing/apiserver.go

```diff
@@ -0,0 +1,198 @@
+// Copyright 2018 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package testing
+
+import (
+	"errors"
+	"fmt"
+	"net"
+	"os/exec"
+	"sync"
+	"time"
+
+	"istio.io/istio/pkg/log"
+
+	"k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset/typed/apiextensions/v1beta1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/rest"
+)
+
+const (
+	//apiServerRepository = "docker.io/ozevren/galley-testing"
+	apiServerRepository = "gcr.io/oztest-mixer/galley-testing"
+	apiServerTag        = "v1"
+	connectionRetries   = 10
+	retryBackoff        = time.Second * 3
+)
+
+var singleton *apiServer
+var lock sync.Mutex
+
+// InitAPIServer starts a container with API server. The handle is kept in a package-internal singleton.
+// This should be called from a test main to do one-time initialization of the API Server.
+func InitAPIServer() (err error) {
```

Comment:
don't use the lockers. use sync.Once() instead. https://golang.org/pkg/sync/#Once